test/test_type_hints.py

```diff
@@ -159,55 +160,42 @@ def test_doc_examples(self):
         """
-        Runs mypy over all files specified in mypy.ini
-        Note that mypy.ini is not shipped in an installed version of PyTorch,
-        so this test will only run mypy in a development setup or in CI.
+        Runs mypy over all files specified in our mypy configs
+        Note that our mypy configs are not shipped in an installed
+        version of PyTorch, so this test will only run mypy in a
+        development setup or in CI.
         """
         def is_torch_mypyini(path_to_file):
             with open(path_to_file, 'r') as f:
                 first_line = f.readline()
 
-            if first_line.startswith('# This is the PyTorch MyPy config file'):
+            name = os.path.basename(path_to_file)
+            if first_line.startswith(f'# This is the PyTorch {name} file'):
                 return True
 
             return False
 
-        test_dir = os.path.dirname(os.path.realpath(__file__))
-        repo_rootdir = os.path.join(test_dir, '..')
-        mypy_inifile = os.path.join(repo_rootdir, 'mypy.ini')
-        if not (os.path.exists(mypy_inifile) and is_torch_mypyini(mypy_inifile)):
-            self.skipTest("Can't find PyTorch MyPy config file")
-
-        import numpy
-        if numpy.__version__ == '1.20.0.dev0+7af1024':
-            self.skipTest("Typeannotations in numpy-1.20.0-dev are broken")
-
-        # TODO: Would be better not to chdir here, this affects the entire
-        # process!
-        with set_cwd(repo_rootdir):
-            (stdout, stderr, result) = mypy.api.run([])
-
-        if result != 0:
-            self.fail(f"mypy failed: {stdout} {stderr}")
-
-    @unittest.skipIf(not HAVE_MYPY, "need mypy")
-    def test_run_mypy_strict(self):
-        """
-        Runs mypy over all files specified in mypy-strict.ini
-        """
-        test_dir = os.path.dirname(os.path.realpath(__file__))
-        repo_rootdir = os.path.join(test_dir, '..')
-        mypy_inifile = os.path.join(repo_rootdir, 'mypy-strict.ini')
-        if not os.path.exists(mypy_inifile):
-            self.skipTest("Can't find PyTorch MyPy strict config file")
-
-        with set_cwd(repo_rootdir):
-            (stdout, stderr, result) = mypy.api.run([
-                '--config', mypy_inifile,
-            ])
-
-        if result != 0:
-            self.fail(f"mypy failed: {stdout} {stderr}")
+        for ini in config_files():
+            with self.subTest(msg=ini):
+                test_dir = os.path.dirname(os.path.realpath(__file__))
+                repo_rootdir = os.path.join(test_dir, '..')
+                mypy_inifile = os.path.join(repo_rootdir, ini)
+                if not (os.path.exists(mypy_inifile) and is_torch_mypyini(mypy_inifile)):
+                    self.skipTest("Can't find PyTorch MyPy config file")
+
+                import numpy
+                if numpy.__version__ == '1.20.0.dev0+7af1024':
+                    self.skipTest("Typeannotations in numpy-1.20.0-dev are broken")
```

Comment:
Does `skipTest` here skip just the individual `subTest`, or the entire test? If the latter, is that a problem?

Reply:
https://blog.ganssle.io/articles/2020/04/subtests-in-python.html indicates that it's meant to skip just the subtest, but actually skips everything.

Either way, that doesn't matter much here I think, skipping the whole test is fine (why would there be a difference between mypy config files?). 

Also, the version comparison is clearly wrong - it's only true for a numpy dev build from one particular commit. Can you change it to `numpy.__version__ [:11] == '1.20.0.dev0'`?

Note that numpy 1.20.0 will be out in a few days, and that will ship with fairly complete type annotations and a `py.typed` file.